### PR TITLE
[lit-html] IE11 support

### PR DESCRIPTION
**Pull request by justinfagnani**
Adds IE11 support to lit-html.

Currently, WTR is doing a good job of transpiling our JS for IE and injecting needed JS polyfills. ChildNode#remove and ParentNode#append polyfills are added to a tests-only polyfills.js file, in anticipation of them being added to the webcomponents polyfills.

With that and a couple of changes to tests with `<script>` tags (add a type so they're not parsed as JS) and a change to attribute handling, we have 100% passing on lit-html_test.js.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,8 +2,11 @@
 
 /packages/*/node_modules/
 /packages/*/lib/
-/packages/*/test/
-/packages/*/development/
+/packages/*/test
+/packages/*/development/**/*.js
```
Comment (aomarks) [inline .gitignore:6]: FYI you can also do this (but I don't have a preference):

```
/packages/*/development/
!/packages/*/development/**/*.html
```

File: packages/lit-html/package.json [hunk 2]
```diff
@@ -33,7 +33,7 @@
     "build:ts:watch": "tsc --watch",
     "dev": "scripts/dev.sh",
     "test": "npm run test:dev && npm run test:prod",
-    "test:dev": "cd ../tests && npx wtr '../lit-html/development/**/*_test.js'",
+    "test:dev": "cd ../tests && npx wtr '../lit-html/development/test/**/lit-html_test.js'",
```
Comment (aomarks) [inline packages/lit-html/package.json:36]: Did you mean to include this change?

File: packages/lit-html/src/test/lit-html_test.ts [hunk 3]
```diff
@@ -33,6 +33,47 @@ import {
 const ua = window.navigator.userAgent;
 const isIe = ua.indexOf('Trident/') > 0;
 
+(function (arr) {
```
Comment (aomarks) [inline packages/lit-html/src/test/lit-html_test.ts:36]: Put these polyfills in their own file?

**Review by aomarks — APPROVED** [review]

File: packages/lit-html/src/lit-html.ts [hunk 4]
```diff
@@ -505,14 +518,16 @@ class Template {
...
-              (node as Element).removeAttribute(name);
-              i--;
+              attrsToRemove.push(name);
               this._parts.push({
                 _type: ELEMENT_PART,
                 _index: nodeIndex,
               });
             }
           }
+          for (const name of attrsToRemove) {
```
Comment (kevinpschaaf) [inline packages/lit-html/src/lit-html.ts:528]: Is mutating attributes while iterating them only bad on IE? Add comment to that effect?

File: packages/lit-html/src/lit-html.ts [hunk 5]
```diff
@@ -744,7 +759,7 @@ export class NodePart {
       // set its value, rather than replacing it.
       (node as Text).data = value as string;
     } else {
-      this._commitNode(new Text(value as string));
+      this._commitNode(document.createTextNode(value as string));
```
Comment (kevinpschaaf) [inline packages/lit-html/src/lit-html.ts:762]: Is it worth including this as a dom shim?

File: packages/lit-html/src/test/polyfills.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      enumerable: true,
+      writable: true,
+      value: function remove() {
+        this.parentNode.removeChild(this);
+      }
+    });
+  });
+})([Element.prototype, CharacterData.prototype, DocumentType.prototype]);
+
+// Source: https://github.com/jserz/js_piece/blob/master/DOM/ParentNode/append()/append().md
```
Comment (kevinpschaaf) [inline packages/lit-html/src/test/polyfills.ts:18]: Link 404'ed; just wanted to double-check license.

**Review by kevinpschaaf — CHANGES_REQUESTED** [review]

File: packages/lit-html/src/lit-html.ts (hunk 4 above)
Comment (justinfagnani) [reply packages/lit-html/src/lit-html.ts:528]: done

File: packages/lit-html/src/lit-html.ts (hunk 5 above)
Comment (justinfagnani) [reply packages/lit-html/src/lit-html.ts:762]: I don't think so... it's such a small change in only one place.

File: packages/lit-html/src/test/polyfills.ts (hunk 6 above)
Comment (justinfagnani) [reply packages/lit-html/src/test/polyfills.ts:18]: That link was copied with the code from MDN. I'll link to MDN instead.

File: packages/lit-html/src/test/lit-html_test.ts (hunk 3 above)
Comment (justinfagnani) [reply packages/lit-html/src/test/lit-html_test.ts]: done

**Review by kevinpschaaf — APPROVED** [review]
LGTM

**Pull request merged**